LibTrixi.jl/src/api_c.jl

```diff
@@ -461,6 +461,53 @@ trixi_load_primitive_vars_cfptr() =
     @cfunction(trixi_load_primitive_vars, Cvoid, (Cint, Cint, Ptr{Cdouble}))
 
 
+"""
+    trixi_store_in_database(data::Ptr{Cdouble}, size::Cint, index::Cint,
+                            simstate_handle::Cint)::Cvoid
```

Comment:
Why does the user have to provide the `index`? Wouldn't it be nicer if one could just dump data into the database (or container (or data storage (or data container (or data lake)))) and retrieve the index from the function, just like we do with simstate?

Reply:
This is related to a general flaw of the approach: 

- The data registry has to be created in `init_simstate` of the libelixir, which is called when the simulation is initialized
- `init_simstate` has to be called first in order to get a simulation handle
- Only then, data vectors can be registered
- However, during initialization of the simulation, Trixi will already call calc_sources, where the data is accessed
- To make this work, some dummy data vector has to be added during initialization and has to be overwritten later

Reply:
OK. This sounds really not like an optimal solution. But it's the one we currently ahve, so let's see how far we can take it 😅.